Client/Coordinators/QRCode/QRCodeCoordinator.swift

```diff
@@ -35,6 +35,5 @@ class QRCodeCoordinator: BaseCoordinator, QRCodeDismissHandler {
 
     func dismiss(_ completion: (() -> Void)?) {
         router.dismiss(animated: true, completion: completion)
-        parentCoordinator?.didFinish(from: self)
```

Comment:
For this change, can we make sure all the QR workflow works as intended still? Here are the places:
- From the onboarding, when trying to sign in to sync
- From the iOS homepage, long press on the app icon and scan a QR code
- From the URL bar, scan a QR code

I'm kind of thinking the QRCodeCoordinator won't be removed otherwise from the child array

Reply:
After syncing with @lmarceau I will revert this change because QRCodeCoordinator doesn't get removed from the childCoordinators array which could cause side effects in other flows. A proper fix will be to implement FxA signin flow with Coordinator in another task.
We can track a known issue for Tab Tray refactor where the QR code flow is dismissed but the email path still works so we are not completely blocked